chore(filter): Add a switch statement

danieljbruce opened the pull request:
In the filter function a switch statement is added so that it is clear to the reader what happens when the arguments are of various lengths.

Fixes #1096

danieljbruce committed:
Add a switch statement

In the filter function a switch statement is added so that it is clear to the reader what happens when the arguments are of various lengths

danieljbruce committed:
chore(filter): Add a switch statement

danieljbruce committed:
Merge branch '1096-filter-function-should-have-a-switch' of https://github.com/danieljbruce/nodejs-datastore into 1096-filter-function-should-have-a-switch

vicb commented on src/query.ts:
```diff
@@ -222,24 +222,39 @@ class Query {
+      case 1: {
+        if (isFilter(propertyOrFilter)) {
+          this.entityFilters.push(propertyOrFilter);
+        } else {
```
I think the `else` branch should be removed here (`operatorOrValue` and `value` are not provided)

danieljbruce replied:
ok

vicb commented on src/query.ts:
```diff
@@ -222,24 +222,39 @@ class Query {
+      case 2: {
+        this.filters.push({
+          name: (propertyOrFilter as String).trim(),
+          op: '=' as Operator,
```
`as Operator` is probably not needed here?

danieljbruce replied:
yup

vicb reviewed (approved):
LGTM with a couple nits

danieljbruce committed:
code review suggestions

Remove the else block, it should never happen anyway because if it does then the user is misusing the function

vicb reviewed (approved):
Nice!

danieljbruce committed:
Merge branch 'main' into 1096-filter-function-should-have-a-switch

danieljbruce committed:
Merge branch 'main' into 1096-filter-function-should-have-a-switch

danieljbruce committed:
Merge branch 'main' into 1096-filter-function-should-have-a-switch
